geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ServerConnectionFactory.java

```diff
@@ -54,17 +53,18 @@ private synchronized void initializeAuthenticatorsMap() {
     authenticators = tmp;
   }
 
-  private synchronized ClientProtocolMessageHandler initializeMessageHandler(
+  private synchronized ClientProtocolService initializeClientProtocolService(
       StatisticsFactory statisticsFactory, String statisticsName) {
-    if (protocolHandler != null) {
-      return protocolHandler;
+    if (clientProtocolService != null) {
+      return clientProtocolService;
     }
 
-    ClientProtocolMessageHandler tempHandler = new MessageHandlerFactory().makeMessageHandler();
-    tempHandler.initializeStatistics(statisticsName, statisticsFactory);
+    // use temp to make sure we publish properly.
+    ClientProtocolService tmp = new ClientProtocolServiceLoader().loadService();
```

Comment:
How do we handle ServiceLoaderExceptions here? I can see we handle them in the `findStreamAuthenticator` method. But I believe we should handle them here.

Reply:
Added catch to the factory, which will rethrow as `IOException` so that the `AcceptorImpl` closes the connection properly.